torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -467,9 +467,35 @@ def fixup_branch_inps(graph, add_after, new_args, suffix) -> None:
             + list(true_lifted_freevars.keys())
             + list(false_lifted_freevars.keys()),
         )
-        # TODO: assert that the true/false return values are
-        # consistent
-        example_value = true_r.as_proxy().node.meta["example_value"]
+        true_example_value = true_r.as_proxy().node.meta["example_value"]
+        false_example_value = false_r.as_proxy().node.meta["example_value"]
+
+        def check_tensor_meta_same(a, b):
+            size_a = a.size()
+            size_b = b.size()
+            if len(size_a) != len(size_b):
+                raise UncapturedHigherOrderOpError(
+                    f"Expect branches to return tensor that has same size but got\
+                     true_branch: {size_a}, false_branch: {size_b}",
+                )
+            for l, r in zip(size_a, size_b):
+                if l != r:
+                    raise UncapturedHigherOrderOpError(
+                        f"Expect branches to return tensor with same size but got\
+                         true_branch: {size_a}, false_branch: {size_b}",
+                    )
+            if a.dtype != b.dtype:
+                raise UncapturedHigherOrderOpError(
+                    f"Expect branches to return tensor with same dtype but got\
+                     true branch: {a.dtype}, false_branch: {b.dtype}",
+                )
+            if a.requires_grad != b.requires_grad:
+                raise UncapturedHigherOrderOpError(
+                    f"Expect branches to return tensor with same requires_grad but got\
+                     true branch: {a.requires_grad}, false_branch: {b.requires_grad}",
+                )
+
+        check_tensor_meta_same(true_example_value, false_example_value)
```

Comment:
Can you add a check for stride?

Reply:
In the future we'll have "stride agnostic IR", which will really help the stride/storage_offset situation, but until then...